### ISSUE TITLE
Out-of-memory causes SIGILL

### ISSUE THREAD

**Issue by Noctune** [rust-lang/rust#21592]
This snippet: ([Playpen link](http://is.gd/KI4QhD))
```
#![allow(unstable)]

extern crate test;

fn main() {
    let a: Vec<u8> = Vec::with_capacity(::std::usize::MAX);
    ::test::black_box(a);
}
```

Causes the program to close with a SIGILL (Illegal instruction) error. I would have expected something like a SIGSEGV instead.

**Comment by sfackler** [comment]
This is working as intended. Rust calls the LLVM [trap](http://llvm.org/docs/LangRef.html#llvm-trap-intrinsic) intrinsic which is compiled to the `ud2` instruction which in turn generates a SIGILL, just like [abort()](http://linux.die.net/man/3/abort) does.

**Comment by Noctune** [comment]
But `oom()` calls `abort()`. Shouldn't it use SIGABRT instead? To me, SIGILL makes it seem like the compiler is at fault, which it clearly isn't.

**Comment by kmcallister** [comment]
This does confuse people.  We could override LLVM's choice if we want to.  A standard SIGABRT handler might be nice.

**Comment by alexcrichton** [comment]
Closing as a dupe of https://github.com/rust-lang/rust/issues/14674, but thanks for the issue!

**Comment by BenWiederhake** [comment]
How is this a duplicate?  The other issue is about OOM not being printed *on stackoverflow*, which seems to be fixed.
OOM *still* causes SIGILL, which is very confusing indeed, and should be fixed somehow.  So I'd say this is still an "open" issue.

Maybe `oom_handler` (which for my system resides in `src/libstd/sys/unix/mod.rs`) should also call to `::libc::abort()` instead of `::intrinsics::abort`, just like in the [fix for #31273](https://github.com/rust-lang/rust/commit/6e455640950c4e8cb8bff919edb4854fcf7773ca)?

**Comment by sfackler** [comment]
The word "stack" does not appear in #14674, unless I'm missing somethign.
